### PR TITLE
508 implement utility to enable disable user activity

### PR DESCRIPTION
**Pull request by ivandzen**
Added subdirectory utils, added scripts to mint allowance/denial tokens

### CODE REVIEW

File: evm_loader/utils/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Permissions management scripts
+
+Scripts mint_permission_token.sh mint_many.sh are suited to operate with client's and contract's permission token balances - to grant or deprive permissions.
+
+## Requirements
+
+To use this scripts one should have next CLI utilities installed:
+    - solana CLI interface
+    - neon-cli
```
Comment (apetrovskiy) [inline evm_loader/utils/README.md:9]: I noticed that we don't even have any install/usage instruction for neon-cli.
Alexander N. confirms this, saying also what he'd like to add neon-cli to our docs

It's not an issue of this PR, just be aware

File: evm_loader/utils/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Requirements
+
+To use this scripts one should have next CLI utilities installed:
+    - solana CLI interface
+    - neon-cli 
+    - spl-token program
+
+Non-zero SOL balance is needed on selected Solana network on default account ~/.config/solana/id.json
+Also, user should have correct Mint-authority JSON file
```
Comment (apetrovskiy) [inline evm_loader/utils/README.md:13]: The same, we don't have and I believe never had any mention about this file
Alexander N. mentioned some folder (?)

Probably, here we need to mention where this fail could be withdrawn as a template or how to create it

File: evm_loader/utils/mint_permission_token.sh [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+mint_denial_token() {
+  NEON_ADDRESS=$1
+  MINMAL_BALANCE=$2
+  calc_permission_tokens_diff $NEON_ADDRESS
+  
+  if [ "$DIFFERENCE" -ge "$MINMAL_BALANCE" ]; then
+    MINT_AMOUNT=$(($DIFFERENCE - $MINMAL_BALANCE + 1))
+    echo "Minting $MINT_AMOUNT denial tokens to $NEON_ADDRESS"
+    spl-token mint --url http://localhost:8899 --mint-authority "$MINT_AUTHORITY_FILE" "$NEON_PERMISSION_DENIAL_TOKEN" "$MINT_AMOUNT" -- "$DENIAL_TOKEN_ACCOUNT"
```
Comment (s-medvedev) [inline evm_loader/utils/mint_permission_token.sh:101]: Is it the correct Solana address?

File: evm_loader/utils/mint_permission_token.sh [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+mint_allowance_token() {
+  NEON_ADDRESS=$1
+  MINMAL_BALANCE=$2
+  calc_permission_tokens_diff $NEON_ADDRESS
+
+  if [ "$DIFFERENCE" -lt "$MINMAL_BALANCE" ]; then
+    MINT_AMOUNT=$(($MINMAL_BALANCE - $DIFFERENCE))
+    echo "Minting $MINT_AMOUNT allowance tokens to $NEON_ADDRESS"
+    spl-token mint --url http://localhost:8899 --mint-authority "$MINT_AUTHORITY_FILE" "$NEON_PERMISSION_ALLOWANCE_TOKEN" "$MINT_AMOUNT" -- "$ALLOWANCE_TOKEN_ACCOUNT"
```
Comment (s-medvedev) [inline evm_loader/utils/mint_permission_token.sh:115]: Here too...

File: evm_loader/utils/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+mint_permission_token.sh operates with permission tokens balances of a signle client/contract. Script will estimate current state of client's/contract's permission tokens balances and mint selected tokens if needed. Here is common form of calling this script:
+    > mint_permission_token.sh <solana_url> <evm_loader_id> <mint_authority_json_file> <allow|deny> <client|contract> <neon_eth_address>
+
+Arguments:
+    - solana_url - URL of Solana RPC endpoing (also, default replacements supported: mainnet-beta, testnet, devnet, localhost)
+    - evm_loader_id - address of solana account where EVM is deployed to selected Solana network
+    - mint_authority_json_file - path to JSON file on the local filesystem where private key to mint-authority stored
+    - allow|deny - keyword determining what action should be performed on client's/contract's permissions
+    - client|contract - keyword determining what type of account (client/contract) actually passed to script
+    - neon_eth_address - NEON accoutn address (Eth-compatible - 0x0f45....)
```
Comment (apetrovskiy) [inline evm_loader/utils/README.md:26]: which account is this address of? The account, which is to be given grant/deny permission?

type: accoutn

File: evm_loader/utils/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    - solana CLI interface
+    - neon-cli 
+    - spl-token program
+
+Non-zero SOL balance is needed on selected Solana network on default account ~/.config/solana/id.json
+Also, user should have correct Mint-authority JSON file
+
+## Permission management for a single client/contract
+
+mint_permission_token.sh operates with permission tokens balances of a signle client/contract. Script will estimate current state of client's/contract's permission tokens balances and mint selected tokens if needed. Here is common form of calling this script:
```
Comment (apetrovskiy) [inline evm_loader/utils/README.md:17]: I wouldn't use a 'blockchain' term for logically non-blockchain thing: if we set allow/deny permissions, names of scripts could be specific, like, for example:
```
set_single_permission.sh
set_multiple_permissions.sh
```

File: evm_loader/utils/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    - client|contract - keyword determining what type of account (client/contract) actually passed to script
+    - neon_eth_address - NEON accoutn address (Eth-compatible - 0x0f45....)
+
+## Permission management for several clients/contracts
+
+mint_many.sh operates with permission tokens balances of several clients/contracts. Here is common form of calling this script:
+    > mint_many.sh <solana_url> <evm_loader_id> <mint_authority_json_file> <allow|deny> <client|contract> <address_list_file>
+
+This script accepts almost the same set of argument except the last one:
+    - address_list_file - path to the file on a local filesystem containing list of NEON account addresses written one on a line (without any punktuation marks)
```
Comment (apetrovskiy) [inline evm_loader/utils/README.md:34]: as in the header we use 'clients/contracts', I'd use the same instead of 'account'

Comment (apetrovskiy) [reply evm_loader/utils/README.md:34]: written one by one in one column
or
written each one on a next line

File: evm_loader/utils/README.md (hunk 2 above)
Comment (ivandzen) [reply evm_loader/utils/README.md:13]: This file created automatically when installing solana it is not related to Neon-EVM directly

File: evm_loader/utils/mint_permission_token.sh (hunk 3 above)
Comment (ivandzen) [reply evm_loader/utils/mint_permission_token.sh:101]: Thanks alot! Silly error =)

File: evm_loader/utils/mint_permission_token.sh (hunk 4 above)
Comment (ivandzen) [reply evm_loader/utils/mint_permission_token.sh:115]: Thanks alot! Silly error =)

File: evm_loader/utils/README.md (hunk 1 above)
Comment (ivandzen) [reply evm_loader/utils/README.md:9]: I believe, some day we will publish it in [crates.io](https://crates.io/)

File: evm_loader/utils/README.md (hunk 5 above)
Comment (ivandzen) [reply evm_loader/utils/README.md:26]: Fixed

File: evm_loader/utils/README.md (hunk 7 above)
Comment (ivandzen) [reply evm_loader/utils/README.md:34]: Fixed

File: evm_loader/utils/README.md (hunk 6 above)
Comment (ivandzen) [reply evm_loader/utils/README.md:17]: Files renamed

**Review by apetrovskiy — APPROVED** [review]

**Pull request merged**